modules/virusshare.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017 Conix Cybersecurity
+# Copyright (c) 2017 Hicham Megherbi
+#
+# This file is part of BTG.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>.
+
+import re
+
+import requests
+
+from lib.io import module as mod
+
+
+class Virusshare:
+    def __init__(self, ioc, type, config):
+        self.config = config
+        self.module_name = __name__.split(".")[1]
+        self.types = ["MD5", "SHA1", "SHA256"]
+        self.search_method = "Online"
+        self.description = "Search IOC malware in VirusShare"
+        self.author = "Hicham Megherbi"
+        self.creation_date = "15-11-2017"
+        self.type = type
+        self.ioc = ioc
+
+        if type in self.types and mod.allowedToSearch(self.search_method):
+            self.search()
+        else:
+            mod.display(self.module_name, "", "INFO", "VirusShare module not activated")
+
+    def serach_ioc(self):
+        search_url = "https://virusshare.com/search.4n6"
+        login_url = "https://virusshare.com/processlogin.4n6"
+        login_page = "https://virusshare.com/login.4n6"
+
+        auth = {'username': self.config['virusshare_username'],
+                'password': self.config['virusshare_password']
+                }
+
+        header = {'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64; rv:52.0) Gecko/20100101 Firefox/52.0',
```

Comment:
User-agent needs to be retrieved from config file : https://github.com/conix-security/BTG/blob/master/config.ini.editme#L15